core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -128,6 +149,25 @@ case class ExceptionFailure(
 }
 
 /**
+ * A class for recovering from exceptions when deserializing a Throwable that was
+ * thrown in user task code. If the Throwable cannot be deserialized it will be null,
+ * but the stacktrace and message will be preserved correctly in SparkException.
+ */
+private[spark] class ThrowableSerializationWrapper(var exception: Throwable) extends
```

Comment:
its a good idea to make this `private[spark]` if we can -- but its actually going to leak as a member of `ExceptionFailure`.  (I'm surprised that scalac doesn't catch this.)  In any case, I think you can fix this by changing to the member of `ExceptionFailure` to `private val exceptionWrapper: Option[ThrowableSerializationWrapper]`.